pkg/secrethub/audit.go

```diff
@@ -32,3 +33,40 @@ func (c *Client) decryptAuditEvents(events ...*api.Audit) error {
 
 	return nil
 }
+
+func newAuditEventIterator(paginator *http.AuditPaginator, client *Client) AuditEventIterator {
+	return AuditEventIterator{
+		iterator: iterator{
+			paginator: paginator,
+		},
+		paginator:          paginator,
+		decryptAuditEvents: client.decryptAuditEvents,
+	}
+}
+
+type AuditEventIterator struct {
+	iterator
+	decryptAuditEvents func(...*api.Audit) error
+	paginator          *http.AuditPaginator
+}
+
+func (it *AuditEventIterator) Next() (api.Audit, error) {
+	item, err := it.next()
+	if err != nil {
+		return api.Audit{}, err
+	}
+	audit := item.(api.Audit)
+	err = it.decryptAuditEvents(&audit)
```

Comment:
Question: we've been noticing the client side decryption of audit events can take `1-3s`  (whereas the server response is `~100ms`). The flow used to be: fetch a page, decrypt the page, show the page. Now it looks like the flow will be: fetch a page, wait til `iterator.Next()` and for each call decrypt that event.

What are your thoughts on the impact on latency of this iterator pattern? Fixing the latency is probably a separate issue and worthy of its own PR, but if this seriously aggravates the latency issue, then we should make a conscious decision about that now.

Reply:
The flow used to be:
1. fetch events
2. sequentially decrypt all events
3. return all events
---
Now it is:
1. fetch page of events
2. decrypt 1 event
3. return 1 event
4. repeat at 2 (or 1 when end of page is reached)

So, when `Next` is called sequentially from 1 go-routine, decryption of events is done sequentially in that go-routine (status quo maintained).

When `Next` is called concurrently, decryption of events is also done concurrently. This reduces the latency problem when multiple cores are available.

We can improve as follows:
1. fetch page of events + concurrently decrypt events
2. wait for 1 event to be decrypted
3. return event
4. repeat at 2 (or 1 when end of page is reached (or when we've buffering implemented, when the buffer is smaller than desired))

Now decryption is always done concurrently, independent of how the caller uses the iterator.